src/ray/core_worker/core_worker.cc

```diff
@@ -1895,12 +1895,31 @@ void CoreWorker::HandlePushTask(const rpc::PushTaskRequest &request,
     return;
   }
 
+  const TaskSpecification task_spec(request.task_spec());
+
+  // Increment the task_queue_length
   task_queue_length_ += 1;
+
+  // For actor tasks, we just need to post a HandleActorTask instance to the task
+  // execution service.
+  if (task_spec.IsActorTask()) {
+    task_execution_service_.post([=] {
+      // We have posted an exit task onto the main event loop,
+      // so shouldn't bother executing any further work.
+      if (exiting_) return;
+      direct_task_receiver_->HandleActorTask(request, reply, send_reply_callback);
+    });
+    return;
```

Comment:
Nit, but could you avoid using early-return here and just put the non-actor codepath in the `else` branch?